src/models/ObjectLevelCan.ts

```diff
@@ -1,17 +1,21 @@
-type BasicPermissions = 'create' | 'read' | 'update' | 'delete'
-
-
 type ObjectLevelCan<TPermissions extends string> = {
   [key in TPermissions]: boolean
 }
 
-export type BasicPermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions>
-export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions | 'run'>
+export const BasicObjectLevelPermissions = ['create', 'read', 'update', 'delete'] as const
+export const RunnableObjectLevelPermissions = [...BasicObjectLevelPermissions, 'run'] as const
+
+export type BasicPermissionsObjectLevelCan = ObjectLevelCan<typeof BasicObjectLevelPermissions[number]>
+export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<typeof RunnableObjectLevelPermissions[number]>
 
-export function createObjectLevelCan<T extends Record<string, boolean>>(): T {
+export function createObjectLevelCan<T extends Record<string, boolean>>(knownProperties: typeof BasicObjectLevelPermissions | typeof RunnableObjectLevelPermissions): T {
+  const set = new Set(knownProperties)
```

Comment:
I don't have a perf to point to ATM but I've had issues in the past where creating lots of sets was actually a lot slower than using an array. Since this happens for every single deployment/block and considering how small these array are IMO we should just check the array directly using `Array.includes` which I actually think is faster on super small arrays like this.

Reply:
oh i forgot that each object invokes this on creation as opposed to a single can for deployments.